### DISCUSSION TITLE
Triggering an old image caching bug.

### DISCUSSION THREAD

**Discussion by Suges** [cytoscape/cytoscape.js#2920] [Q&A] [answered]
Hi and big thanks to Max for creating this project!

In [this](https://stackoverflow.com/questions/35770055/replace-all-elements-and-redraw-graph-softly-in-cytoscape-js) article Max says to clear out and reload a graph (which is necessary to do if the _new_ graph has some of the same element IDs as the old one), the thing to do is: `cy.elements().remove(); cy.add( newEleJsons );`.

But I've noticed (at least on the latest versions of Chrome and Safari) doing that seems to trigger this old chestnut of a bug: https://github.com/cytoscape/cytoscape.js/issues/1511 (exact same symptoms: only some of the pictures show up; if you click on the node it makes the other show up--but then some of the originals disappear).

You can always fix it by calling something like: `const x = cy.json(); cy.json({ elements: x });`, but I thought I'd just bring it up here if anyone's interested.

**Comment by maxkfranz** [comment]
Could you put together a small demo of your issue?  There’s a jsbin demo that you can fork to easily put something together

http://jsbin.com/fiqugiq

**Comment by Suges** [comment, marked as answer]
I think I actually just squashed the bug while getting ready to transfer it to jsbin.

I accidentally had _two_ event listeners on my "load a drawing" button; `cy.elements().remove(); cy.add( newEleJsons )` was getting triggered twice in rapid succession. Since I removed the second trigger Cytoscape settled right down.

I think we're good here, thanks for your response!
